src/dlls/mscoree/unixinterface.cpp

```diff
@@ -88,48 +88,30 @@ static LPCWSTR* StringArrayToUnicode(int argc, LPCSTR* argv)
 }
 
 //
-// Execute a managed assembly with given arguments
+// Initialize the CoreCLR. Creates and starts CoreCLR host and creates an app domain
 //
 // Parameters:
 //  exePath                 - Absolute path of the executable that invoked the ExecuteAssembly
-//  coreClrPath             - Absolute path of the libcoreclr.so
 //  appDomainFriendlyName   - Friendly name of the app domain that will be created to execute the assembly
 //  propertyCount           - Number of properties (elements of the following two arguments)
 //  propertyKeys            - Keys of properties of the app domain
 //  propertyValues          - Values of properties of the app domain
-//  argc                    - Number of arguments passed to the executed assembly
-//  argv                    - Array of arguments passed to the executed assembly
-//  managedAssemblyPath     - Path of the managed assembly to execute (or NULL if using a custom entrypoint).
-//  enntyPointAssemblyName  - Name of the assembly which holds the custom entry point (or NULL to use managedAssemblyPath).
-//  entryPointTypeName      - Name of the type which holds the custom entry point (or NULL to use managedAssemblyPath).
-//  entryPointMethodName    - Name of the method which is the custom entry point (or NULL to use managedAssemblyPath).
-//  exitCode                - Exit code returned by the executed assembly
+//  hostHandle              - Output parameter, handle of the created host
+//  domainId                - Output parameter, id of the created app domain 
 //
 // Returns:
 //  HRESULT indicating status of the operation. S_OK if the assembly was successfully executed
 //
 extern "C"
-HRESULT ExecuteAssembly(
+HRESULT InitializeCoreCLR(
             LPCSTR exePath,
-            LPCSTR coreClrPath,
```

Comment:
It would be nice to only use portable C types in the signatures of these methods (char*, int, int32_t, etc.) so that the hosts are not getting polluted by Windows specific goo.